src/core/thread/topology.hpp

```diff
@@ -664,7 +664,7 @@ class Neighbor : public InstanceLocatorInit
     void SetTimeSyncEnabled(bool aEnabled) { mTimeSyncEnabled = aEnabled; }
 #endif
 
-#if OPENTHREAD_CONFIG_MLE_LINK_METRICS_ENABLE
+#if OPENTHREAD_CONFIG_MLE_LINK_METRICS_INITIATOR_ENABLE || OPENTHREAD_CONFIG_MLE_LINK_METRICS_SUBJECT_ENABLE
```

Comment:
needed in both unfortunatelly

Reply:
The concept here is a little confusing. Most methods here are only used by Subjects. But  `GetEnhAckProbingMetrics` and `SetEnhAckProbingMetrics` are used by Initiators. I think we can separate the two methods from others.

Let me explain a little for the two methods:
When an Initiator configures Enh-ACK probing, it will get Vendor Specific IE from a Subject's Enh ACK. But the data in the IE only contains values and doesn't have types. So the Get/Set method records for each Subject Neighbor what metrics did the Initiator configure so that the Initiator know what these data are.